### PR TITLE
games-emulation/pcsx2: fixed compilation with gcc 5

### PR DESCRIPTION
**Pull request by nE0sIghT**
Gentoo-Bug: https://bugs.gentoo.org/show_bug.cgi?id=605162

### CODE REVIEW

**Comment by nvinson** [comment]
Could you change
```
Gentoo-Bug: https://bugs.gentoo.org/show_bug.cgi?id=605162
```
in your commit to
```
Gentoo-bug: 605162
```
please?  The commit will align better with the [Gentoo Developers Manual](https://devmanual.gentoo.org/ebuild-maintenance/index.html#git-commit-message-format) that way.

**Comment by a17r** [comment]
This is a build fix, so it does not really need a revbump.

**Comment by nE0sIghT** [comment]
@nvinson
Gentoo's devmanual allows to use both forms: just bug num or bugzilla URI. I do not see any note about short form preference against URI. Furthermore I see both forms are used in recent tree commits.
I personally prefer URI because it's just clickable in GUIs. Could you please explain why short form is preferable?

@a17r
Is it allowed to bump EAPI in that case or should I drop EAPI part?

**Comment by nE0sIghT** [comment]
@a17r 
There is also [following notice in devmanual](https://devmanual.gentoo.org/general-concepts/ebuild-revisions/):

> Important: For ebuilds marked stable on at least one arch, only trivial edits can be made without a bump (e.g. typo fixes in elog messages). Even simple changes may result in a breakage. Modifying stable ebuilds should be avoided.

**Comment by a17r** [comment]
Also from your link:

> Simple compile fixes do not warrant a revision bump; this is because they do not affect the installed package for users who already managed to compile it.

But I don't know how many changes you had to make for the EAPI 6 bump.

File: games-emulation/pcsx2/pcsx2-1.4.0-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+# Ensure no incompatible headers from eselect-opengl are installed, bug #510730
+DEPEND="${RDEPEND}
+	>=app-eselect/eselect-opengl-1.3.1
+	dev-cpp/pngpp
+	>=dev-cpp/sparsehash-1.5
+"
+
+S="${WORKDIR}/${PN}-${MY_PV}"
+
+PATCHES="${FILESDIR}"/"${P}"-gcc5.patch
```
Comment (a17r) [inline games-emulation/pcsx2/pcsx2-1.4.0-r1.ebuild]: `PATCHES` should be an array.

**Comment by nE0sIghT** [comment]
I'm think it's a bad idea to touch stable package with EAPI change and patch addition.
Even if there is no ebuild changes besides EAPI bump - it's not just a number change.

I would like to get one more review for this.

File: games-emulation/pcsx2/pcsx2-1.4.0-r1.ebuild (hunk 1 above)
Comment (nE0sIghT) [inline games-emulation/pcsx2/pcsx2-1.4.0-r1.ebuild]: As I see implementation [allows strings](https://devmanual.gentoo.org/ebuild-writing/eapi/index.html#eapi=6).
Is it bad usage?

Comment (a17r) [inline games-emulation/pcsx2/pcsx2-1.4.0-r1.ebuild]: It does allow strings, but using an array is *highly encouraged*. This was common to all my reviews and is also mentioned in https://blogs.gentoo.org/mgorny/2015/11/13/the-ultimate-guide-to-eapi-6/

Comment (nE0sIghT) [inline games-emulation/pcsx2/pcsx2-1.4.0-r1.ebuild]: Thanks for link. I will push new commit soon

**Comment by a17r** [comment]
If you are mostly concerned about the EAPI change, then you could leave that to the 9999 ebuild.

**Comment by nE0sIghT** [comment]
I dropped EAPI bump, revbump and changed PATCH to array

**Comment by nvinson** [comment]
> Gentoo's devmanual allows to use both forms: just bug num or bugzilla URI. I do not see any note about short form preference against URI. Furthermore I see both forms are used in recent tree commits.
> I personally prefer URI because it's just clickable in GUIs. Could you please explain why short form is preferable?

Because I missed the part that said the bug ID could be the URI.  On a personal note, I find having the full URI to be unnecessarily noisy.

That said, personal preference isn't enough to request changes (otherwise there would be no end to the requests), so I'll retract my request that the commit message be updated.

**Comment by nE0sIghT** [comment]
Ping

**Pull request closed without merging**